### DISCUSSION TITLE
DeskModder discussion

### DISCUSSION THREAD

**Discussion by rlnt** [Sophia-Community/SophiApp#32] [General]
Hello!

DeskModder once again had an article about SophiApp. I know you've seen it because I read all the comments.
To close the gap caused by the language barrier of the article, I would like to start a discussion here.
I am a native German speaker and follower of the DeskModder blog. I am also a long-time SophiScript user and am currently waiting for Windows 11 support of the SophiApp. The script was always a big help when setting up new systems. So thanks for that!

I think the main arguments of people replying [here](https://www.deskmodder.de/blog/2022/02/07/sophiapp-einmal-angeschaut/#comments) were the UI of the app and the translation.

About the translation:
I know you're offering that people can translate the app. I can submit a German translation but I have one question beforehand.
Is a German translation in the `UIData.json` needed if the German string would match the English string? I am used to XML string tables or Fluent. Usually, if a specific language doesn't have an entry for a string, it will fall back to English. If the string of the German language matches the English one, a translation wouldn't be necessary.

About the UI:
I am not a big fan of the UI either. It's a pretty ugly app (sorry). But that's just my personal opinion and it's my personal preference. But I see some of the people in the comments agree. This shouldn't be much of an issue though because, in the end, it's the app's functionality that matters. Is a UI rework planned in the future? I wouldn't mind if you just focus on functionality, bug fixing and most importantly OS support.

**Comment by farag2** [comment]
Hello! Glad to see you here, and we appreciate your effort to help with the translation.

Let’s take the first function that has ID 100: <https://github.com/Sophia-Community/SophiApp/blob/8461c834e53941b23837c577086ca5acf115afda/SophiApp/SophiApp/Resources/UIData.json#L3>

All you need it to add new „DE“ strings. Look:

```JSON
{
  "Id": 100,
  "Header": {
    "EN": "The telemetry service",
    "RU": "Служба телеметрии",
    "DE": "Der Telemetrie-Dienst"
  },
  "Description": {
    "EN": "Configure the Connected User Experiences and Telemetry (DiagTrack) service, and connection for the Unified Telemetry Client Outbound Traffic",
    "RU": "Настроить cлужбу \"Функциональные возможности для подключенных пользователей и телеметрия\" (DiagTrack) и соединение для исходящего трафика клиента единой телеметрии",
    "DE": "Konfigurieren Sie den Dienst Connected User Experiences and Telemetry (DiagTrack) und die Verbindung für den ausgehenden Verkehr des Unified Telemetry Client"
  },
  "Type": "CheckBox",
  "Tag": "Privacy"
}
```


As I wrote in the comments, we've understood that if the file grows bigger it will look like a mess, so need to split it into languages to make the traslation process easier. Unfortuanately, we aren't confident that the current description sections is finished. I make changes to it from time to time. I think in a month or so I'll finish it up, and you may start translating.

> Usually, if a specific language doesn't have an entry for a string, it will fall back to English.

You're right, but in this case, if there is no translatable description, the entry will be empty,because the parse engine wasn't able to find it. Every translation has to have its' separate file to parse.

>  It's a pretty ugly app (sorry).

I've heard it for the first time. Seriously )). Well, I have not words. But we've already started working out 2.0. I want SophiApp 2.0 to follow  the Windows 11 UI guidelines. Look at Windows Terminal 1.13 Settings UI or Power Toys:
<https://devblogs.microsoft.com/commandline/windows-terminal-preview-1-13-release/>
<https://www.onmsft.com/news/powertoys-update-brings-new-find-my-mouse-tool>

Does this UI look ugly for your too? 🐱

**Comment by rlnt** [comment]
I am aware of how JSONs work and I know that another language-specific XML is required. I just wasn't sure if a translation is required if it matches the English original phrase. But thanks for the clarification. 👍
You can simply mention me here in this discussion once you think it's worth submitting the German translation! :)

The layout of the current design is alright but the color is really weird. I am also fine with the design of the current Script Wrapper.
But if you want to redesign it like the new PowerToys version of the Windows Terminal (two apps I use as well), that would be lovely. Especially since this grants consistent design with the new OS style. This is very clean and nice.

**Comment by farag2** [comment]
https://github.com/Sophia-Community/SophiApp/pull/20#issuecomment-1083601519
